packages/state-transition/src/block/processBlockHeader.ts

```diff
@@ -29,25 +29,19 @@ export function processBlockHeader(state: CachedBeaconStateAllForks, block: allF
     );
   }
 
-  const types = state.config.getForkTypes(slot);
   // verify that the parent matches
   if (!byteArrayEquals(block.parentRoot, ssz.phase0.BeaconBlockHeader.hashTreeRoot(state.latestBlockHeader))) {
     throw new Error(
       `Block parent root ${toHexString(block.parentRoot)} does not match state latest block, block slot=${slot}`
     );
   }
 
-  const bodyRoot = isBlindedBeaconBlock(block)
-    ? ssz.bellatrix.BlindedBeaconBlockBody.hashTreeRoot(block.body)
-    : types.BeaconBlockBody.hashTreeRoot(block.body);
-
+  const blockHeader = blindedOrFullBlockToHeader(state.config, block as allForks.FullOrBlindedBeaconBlock);
   // cache current block as the new latest block
+  // TODO: do we need to ZERO_HASH the stateRoot here??
   state.latestBlockHeader = ssz.phase0.BeaconBlockHeader.toViewDU({
-    slot: slot,
-    proposerIndex: block.proposerIndex,
-    parentRoot: block.parentRoot,
+    ...blockHeader,
```

Comment:
Would prefer not to spread here, map all properties manually. About ZERO_HASH, yes that's required by the spec